cmux-tui/bindings/reconcile_registry_artifact.py

```
@@ -0,0 +1,307 @@
+        raise RegistryError(f"npm metadata has no dist object for {package}@{version}")
+    integrity = dist.get("integrity")
+    if isinstance(integrity, str) and "-" in integrity:
+        algorithm = integrity.split("-", 1)[0]
+        try:
+            local = _integrity(artifact, algorithm)
+        except ValueError as error:
+            raise RegistryError(
+                "npm returned an unsupported integrity algorithm for "
+                f"{package}@{version}: "
+                f"{algorithm}"
+            ) from error
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)